tests/api/test_simple_archive.cc

```diff
@@ -1,300 +1,129 @@
-    //    // so we should get exceptions raised if anything else is used
-
-    //    dasi::Key key{
-    //            {"key1",  "value1"},
-    //            {"key2",  "123"},
-    //            {"key3",  "value1"},
-    //            {"key1a", "value1"},
-    //            {"key2a", "value1"},
-    //            {"key3a", "blah-blah-nonintegral"},
-    //            {"key1b", "value1"},
-    //            {"key2b", "value1"},
-    //            {"key3b", "value1"},
-    //    };
-
-    //    constexpr const char test_data[] = "TESTING SIMPLE ARCHIVE";
-    //    EXPECT_THROWS_AS(dasi.archive(key, test_data, sizeof(test_data) - 1), eckit::SeriousBug);
-    //}
 }
 
+CASE("Archive data and check list and retrieve") {
+    TempDirectory tempDir;
 
-CASE("Accessing data that has been archived") {
+    simpleWrite(tempDir, "simple_schema", SIMPLE_SCHEMA);
 
-    eckit::TmpDir test_wd(eckit::LocalPathName::cwd().c_str());
-    std::cout << "Working directory: " << test_wd << std::endl;
+    const auto cfg = simpleConfig(tempDir, "simple_schema");
 
-    test_wd.mkdir();
-    (test_wd / "root").mkdir();
+    dasi::Dasi dasi(cfg.c_str());
 
-    {
-        eckit::FileHandle dh(test_wd / "simple_schema");
-        dh.openForWrite(0);
-        eckit::AutoClose close(dh);
-        dh.write(SIMPLE_SCHEMA, sizeof(SIMPLE_SCHEMA));
-    }
+    const auto keys = KeySet({"value3b1", "value3b2", "value3b3"});
+
+    LOG_D("--- ARCHIVE ---");
 
-    dasi::Dasi dasi(simple_config(test_wd).c_str());
-
-    constexpr const char test_data1[] = "TESTING SIMPLE ARCHIVE";
-    constexpr const char test_data2[] = "TESTING SIMPLE ARCHIVE 2222222222";
-    constexpr const char test_data3[] = "TESTING SIMPLE ARCHIVE 3333333333";
-
-    for (const auto& elems : std::vector<std::tuple<const char*, int, const char*>> {
-                {test_data1, sizeof(test_data1)-1, "value1"},
-                {test_data2, sizeof(test_data2)-1, "value2"},
-                {test_data3, sizeof(test_data3)-1, "value3"}}) {
-
-            dasi::Key key{
-                {"key1",  "value1"},
-                {"key2",  "123"},
-                {"key3",  "value1"},
-                {"key1a", "value1"},
-                {"key2a", "value1"},
-                {"key3a", "321"},
-                {"key1b", "value1"},
-                {"key2b", "value1"},
-                {"key3b", std::get<2>(elems)},
-            };
-
-            dasi.archive(key, std::get<0>(elems), std::get<1>(elems));
+    for (auto&& key : keys) {
+        const std::string data = "DASI ARCHIVE TEST DATA " + key.get("key3b");
+        dasi.archive(key, data.data(), data.size());
     }
-    dasi.flush();
 
-    SECTION("We can list all of the data correctly") {
+    dasi.flush();
 
-        dasi::Query query {
-            {"key1",  {"value1"}},
-            {"key2",  {"123"}},
-            {"key3",  {"value1"}},
-        };
+    LOG_D("--- LIST ---");
 
-        ListResultChecker checker{
```

Comment:
I'm missing something here. It looks to me like a whole load of testing functionality is missing. Where has it gone?